vscode-dotnet-runtime-extension/src/extension.ts

```diff
@@ -190,18 +217,69 @@ export function activate(context: vscode.ExtensionContext, extensionContext?: IE
                 eventStream.post(new DotnetAcquisitionMissingLinuxDependencies());
                 await installer.promptLinuxDependencyInstall('Failed to run .NET runtime.');
             }
-        }, issueContext(commandContext.errorConfiguration, 'ensureDependencies'));
+        }, runtimeIssueContextFunctor(commandContext.errorConfiguration, 'ensureDependencies'));
     });
 
     const reportIssueRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.reportIssue}`, async () => {
-        const [url, issueBody] = formatIssueUrl(undefined, issueContext(AcquireErrorConfiguration.DisableErrorPopups, 'reportIssue'));
+        const [url, issueBody] = formatIssueUrl(undefined, runtimeIssueContextFunctor(AcquireErrorConfiguration.DisableErrorPopups, 'reportIssue'));
         await vscode.env.clipboard.writeText(issueBody);
         open(url);
     });
 
+    // Helper Functions
+    function resolveExistingPathIfExists(configResolver : ExtensionConfigurationWorker, commandContext : IDotnetAcquireContext) : Promise<IDotnetAcquireResult | null>
+    {
+        const existingPathResolver = new ExistingPathResolver();
+
+        const existingPath = existingPathResolver.resolveExistingPath(configResolver.getPathConfigurationValue(), commandContext.requestingExtensionId, displayWorker);
+        if (existingPath) {
+            eventStream.post(new DotnetExistingPathResolutionCompleted(existingPath.dotnetPath));
+            return new Promise((resolve) => {
+                resolve(existingPath);
+            });
+        }
+        return new Promise((resolve) => {
+            resolve(null);
+        });
+    }
+
+    function getAcquisitonWorker(isRuntimeWorker : boolean) : DotnetCoreAcquisitionWorker
```

Comment:
nit: `Acquisition` is typo'd here.

Reply:
Thank you very much :) Resolved this feedback and the above. Please see the latest commit and description and let me know if it looks good, its a fairly significant change, small but important to get right